oak_functions/abi/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Type, constant and Wasm host function definitions for the Oak-Functions application
+//! binary interface (ABI).
+
+// Status values exchanged as i32 values across the WasmInterface.
+#[derive(Debug)]
+pub enum OakStatus {
```

Comment:
Worth marking as `#[repr(u8)]` instead of manually converting to / from numbers?

Reply:
I am not sure how exactly this helps. I still need to implement `from_i32`. But an automatic conversion would be nice. I noticed, I had already missed some values.

Reply:
You are right, I think that only allows converting from enum to number, but not the other way around. There are some crates that do the conversion both ways, but not sure it's worth introducing them for now. Also perhaps after https://github.com/project-oak/oak/pull/1959 is merged we can actually use a protobuf enum for this, like for the regular Oak ABI?